src/extensions/replay/sessionrecording.ts

```diff
@@ -81,6 +61,20 @@ const ACTIVE_SOURCES = [
     IncrementalSource.Drag,
 ]
 
+const typesAllowedWhenIdle = [EventType.Custom, EventType.Meta, EventType.FullSnapshot]
+
+/**
+ * we want to restrict the data allowed when we've detected an idle session
+ * but allow data that the player might require for proper playback
+ */
+function allowedWhenIdle(event: eventWithTime): boolean {
+    const isInactiveIncremental =
+        event.type === EventType.IncrementalSnapshot &&
+        !isNullish(event.data.source) &&
+        !ACTIVE_SOURCES.includes(event.data.source)
```

Comment:
I'm assuming active events can happen in an idle state and that's why this is useful

Reply:
it's a little belt and braces logically correct if a little stupid

if we see an active event then we should switch out of idle

but i want this to make sense so if we change/break things elsewhere we get a signal from the tests

Reply:
I'm leaning towards capturing everything even in an idle state, most likely This is the only reliable way of having everything for playback, I wonder how much more data we were capturing and that became a problem so we implemented the idle state check, no context around that but I wonder if the problem was bigger than the current broken experience for this use case.

Reply:
it's really tricky...

it was things like "why does my session say it's 12 hours long when it was only 2 minutes long"

because you captured 12 hours of a web socket updating the DOM

(we'll be back in that world now btw - i want to see how many of the current playback bugs are fixed by this and then figure out how to have both fixed at the same time - may be that this isn't actually the problem 🙈)

Reply:
made me realise I'm trading one possible problem for one known problem... so I've changed the approach here

Reply:
yeah, understand that, maybe the player has to be clever enough and allow the "skipping inactivity" also to skip when the browser is inactive, or even add a new option like "skip user idle" or something, just thinking out loud tho, let's get this in and see if it works better.